auth_test.go

```diff
@@ -408,6 +409,89 @@ func TestVerifyMessageWithMTPProof_Merkalized(t *testing.T) {
+
+	schemaLoader := &mockJSONLDSchemaLoader{}
+	authInstance, err := NewVerifier(verificationKeyloader, stateResolvers, WithDocumentLoader(schemaLoader))
+	require.NoError(t, err)
+
+	type expected struct {
+		err  bool
+		Typ  iden3comm.MediaType
+		From string
+		To   string
+	}
+	for _, tc := range []struct {
+		name      string
+		token     string
+		verifyOpt pubsignals.VerifyOpt
+		expected  expected
+	}{
+		{
+			name:  "Wrong token",
+			token: "wrongtoken",
+			expected: expected{
+				err: true,
+			},
+		},
+		{
+			name:  "A dummy JWT token",
+			token: "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiaWF0IjoxNTE2MjM5MDIyfQ.SflKxwRJSMeKKF2QT4fwpMeJf36POk6yJV_adQssw5c",
+			expected: expected{
+				err: true,
+			},
+		},
+		{
+			name:  "Verify JWS token",
+			token: tokenJWS,
+			expected: expected{
+				Typ:  packers.MediaTypeSignedMessage,
+				From: "did:pkh:poly:0x7141E4d20F7644DC8c0AdCA8a520EC83C6cABD65",
+				To:   "did:polygonid:polygon:mumbai:2qLPqvayNQz9TA2r5VPxUugoF18teGU583zJ859wfy",
+			},
+		},
+		{
+			name:      "Verify JWZ token, no proof generation delay",
+			token:     tokenJWZPrivadoMain,
+			verifyOpt: nil,
+			expected: expected{
+				Typ:  packers.MediaTypePlainMessage,
+				From: "did:polygonid:polygon:mumbai:2qPDLXDaU1xa1ERTb1XKBfPCB3o2wA46q49neiXWwY",
+				To:   "did:polygonid:polygon:mumbai:2qJ689kpoJxcSzB5sAFJtPsSBSrHF5dq722BHMqURL",
+			},
+		},
+		{
+			name:      "Verify JWZ token, with proof generation delay",
```

Comment:
This test is failing because network inside token is 80001 and there is no resolver for it. 

I guess I need a valid token, or to mock eth.Resolvers. To mock eth.Resolvers, we should create a new interface in iden3comm lib for map[int]eth.Resolvers